### PR TITLE
Rename `replace_sampler_ddp|replace_sampler` to `use_distributed_sampler`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

The name is agnostic to DDP
Fabric and the Trainer now match

cc @justusschock @awaelchli @borda @carmocca

### CODE REVIEW

File: src/lightning/fabric/fabric.py [hunk 1]
```diff
@@ -291,21 +292,24 @@ def setup_dataloaders(
...
     ) -> DataLoader:
         """Set up a single dataloader for accelerated training.
 
         Args:
             dataloader: The dataloader to accelerate.
-            replace_sampler: If set ``True`` (default), automatically wraps or replaces the sampler on the dataloader
-                for distributed training. If you have a custom sampler defined, set this to this argument to ``False``.
+            use_distributed_sampler: If set ``True`` (default), automatically wraps or replaces the sampler on the
+                dataloader for distributed training. If you have a custom sampler defined, set this to this argument to
+                ``False``.
```
Comment (awaelchli) [inline src/lightning/fabric/fabric.py:312]:
```suggestion
                dataloader for distributed training. If you have a custom sampler defined, set this argument to
                ``False``.
```

File: src/lightning/fabric/fabric.py [hunk 2]
```diff
@@ -273,15 +273,16 @@ def setup_optimizers(self, *optimizers: Optimizer) -> Union[_FabricOptimizer, Tu
...
         """Set up one or multiple dataloaders for accelerated training. If you need different settings for each
         dataloader, call this method individually for each one.
 
         Args:
             *dataloaders: A single dataloader or a sequence of dataloaders.
-            replace_sampler: If set ``True`` (default), automatically wraps or replaces the sampler on the dataloader(s)
-                for distributed training. If you have a custom sampler defined, set this to this argument to ``False``.
+            use_distributed_sampler: If set ``True`` (default), automatically wraps or replaces the sampler on the
+                dataloader(s) for distributed training. If you have a custom sampler defined, set this to this argument
+                to ``False``.
```
Comment (awaelchli) [inline src/lightning/fabric/fabric.py:285]:
```suggestion
                dataloader(s) for distributed training. If you have a custom sampler defined, set this argument
                to ``False``.
```

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/common/trainer.rst [hunk 3]
```diff
@@ -1014,43 +1014,28 @@ The pseudocode applies also to the ``val_dataloader``.
...
     def train_dataloader(self):
+        dataset = ...
         # default used by the Trainer
-        sampler = torch.utils.data.distributed.DistributedSampler(dataset, shuffle=True)
+        sampler = torch.utils.data.DistributedSampler(dataset, shuffle=True)
         dataloader = DataLoader(dataset, batch_size=32, sampler=sampler)
         return dataloader
 
-.. note:: For iterable datasets, we don't do this automatically.
-
```
Comment (awaelchli) [inline docs/source-pytorch/common/trainer.rst:1053]: I think this is important

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: docs/source-pytorch/common/trainer.rst (hunk 3 above)
Comment (carmocca) [reply docs/source-pytorch/common/trainer.rst:1053]: It's in the Trainer argument description, which I reference above

**Pull request merged**